openbb_platform/extensions/platform_api/openbb_platform_api/response_models.py

```
@@ -147,3 +147,149 @@ def validate_model(cls, values) -> "PdfResponseModel":
+    data_type : Optional[str]
+        The type of data, which is always "object" for Omni widgets, no need to specify.
+    parse_as : Optional[str]
+        The type of content to parse as. One of "table", "chart", or "text".
+        Attempts to set this automatically based on the content type, but can be overridden.
+
+    Returns
+    -------
+    object
+        Object that conforms to the validated output requirements of the API.
+
+    Example
+    -------
+    >>> from openbb_platform_api.main import app
+    >>> @app.get("/omni_widget", response_model=OmniWidgetResponseModel)
+    >>> async def get_omni_widget():
+    >>>     return {"content": [{"name": "Alice", "age": 30}, {"name": "Bob", "age": 25}]}
+    """
+
+    model_config = ConfigDict(
+        extra="ignore",
+        json_schema_extra={
+            "x-widget_config": {
+                "$.type": "omni",
+            }
+        },
+    )
+
+    content: Any = Field(
+        description="The content to display in the Omni widget.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+    parse_as: Optional[str] = Field(
+        default=None,
+        description="The type of content to parse as. One of 'table', 'chart', or 'text'.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+    data_format: Optional[dict] = Field(
+        default=None,
+        description="Leave this field empty. This is populated by the model_validator.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+
+    @model_validator(mode="after")
+    @classmethod
+    def validate_model(  # pylint: disable=import-outside-toplevel
+        cls, values
+    ) -> "OmniWidgetResponseModel":
+        """Validate the Omni widget content."""
+        import contextlib  # noqa
+        import json
+        import pandas as pd
+
+        content = getattr(values, "content", None)
+
+        if content is None:
```

When we are returning an empty string it casts to none, no?

---

This is for validating what the user is submitting as the results content, and is a check to ensure that it is not initialized with empty content. The type for "content" is Any, so this is an additional test.

---

My point is that if I explicitly want to return an empty string I will fail because it will cast to None and raise an error

---

> My point is that if I explicitly want to return an empty string I will fail because it will cast to None and raise an error

It's recommended to return an error instead of empty data for other widget types, so this just follows the same pattern.

---

> My point is that if I explicitly want to return an empty string I will fail because it will cast to None and raise an error

An empty string is not actually None. Only None can be None.

```
In [1]: content = ""

In [2]: content is None
Out[2]: False
```